### ISSUE TITLE
fcp:doSelectMenu({"Edit", "Undo.*"}) Broken

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#1952]
```lua
> require("cp.apple.finalcutpro"):doSelectMenu({"Edit", "Undo.*"}):Now()

2019-08-26 20:57:41: ********
2019-08-26 20:57:41: 20:57:41 ERROR:   LuaSkin: hs.timer callback error: ...chrishocking/Github/CommandPost/src/extensions/cp/rx.lua:178: Unable to find 'Undo.*' in 'Edit' while in 'en'.
stack traceback:
	[C]: in function 'error'
	...chrishocking/Github/CommandPost/src/extensions/cp/rx.lua:178: in field '?'
	...chrishocking/Github/CommandPost/src/extensions/cp/rx.lua:1157: in function <...chrishocking/Github/CommandPost/src/extensions/cp/rx.lua:1156>
2019-08-26 20:57:41: ********
```

**Comment by latenitefilms** [comment]
Also fails in non-Rx:

```lua
> require("cp.apple.finalcutpro"):selectMenu({"Edit", "Undo.*"})
false
```

**Comment by latenitefilms** [comment]
This is the result from adding the following debugging code to `exactMatch()`:

```lua
local function exactMatch(value, pattern, plain)
    if value and pattern then
        local s,e = value:find(pattern, nil, plain)
        local result = s == 1 and e == value:len()
        log.df("value: %s, pattern: %s, plain: %s, result: %s", value, pattern, plain, result)
        return result
    end
    log.df("value: %s, pattern: %s, plain: %s, result: %s", value, pattern, plain, false)
    return false
end
```

```lua
> require("cp.apple.finalcutpro"):menu():doFindMenuUI({"Edit", "Undo.*"}):Now(function(item) print(item:title() .. " enabled: ", item:enabled()) end, error)
2019-08-27 21:00:13: 21:00:13                 value: Final Cut Pro, pattern: Edit, plain: nil, result: false
2019-08-27 21:00:13:                          value: File, pattern: Edit, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Edit, plain: nil, result: true
2019-08-27 21:00:13:                          value: Edit, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Redo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Cut, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Copy, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Copy Timecode, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Paste, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Paste as Connected Clip, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Delete, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Replace with Gap, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Select All, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Select Clip, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Deselect All, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Select, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Paste Effects, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Paste Attributes…, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Remove Effects, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Remove Attributes…, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Duplicate, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Duplicate Project as Snapshot, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Keyframes, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Connect to Primary Storyline, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Insert, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Append to Storyline, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Overwrite, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Source Media, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Overwrite to Primary Storyline, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Lift from Storyline, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Add Default Transition, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Add Default Video Effect, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Add Default Audio Effect, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Connect Title, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Insert Generator, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Connect Freeze Frame, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Captions, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: , plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Find…, plain: nil, result: false
2019-08-27 21:00:13:                          value: Edit, pattern: Find and Replace Title Text..., plain: nil, result: false
2019-08-27 21:00:13: Edit enabled: 	true
2019-08-27 21:00:13:                          value: Undo, pattern: Undo.*, plain: nil, result: true
2019-08-27 21:00:13:                          value: Undo Enable Clip, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Redo Move, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Cut, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Copy, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Copy Timecode, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Paste, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Paste as Connected Clip, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Delete, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Replace with Gap, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Select All, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Select Clip, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Deselect All, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Select, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Paste Effects, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Paste Attributes…, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Remove Effects, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Remove Attributes…, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Duplicate, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Duplicate Project as Snapshot, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Keyframes, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Connect to Primary Storyline, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Insert, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Append to Storyline, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Overwrite, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Source Media, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Overwrite to Primary Storyline, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Lift from Storyline, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Add Cross Dissolve, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Add Color Board, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Add Channel EQ, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Connect Title, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Insert Generator, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Connect Freeze Frame, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Captions, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Find…, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Find and Replace Title Text..., pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: , pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Start Dictation…, pattern: Undo, plain: nil, result: false
2019-08-27 21:00:13:                          value: Emoji & Symbols, pattern: Undo, plain: nil, result: false
```

...so it seems the `exactMatch()` function is working correctly. It's something AFTER that that's failing in terms of pattern matching, which is... odd.

**Comment by latenitefilms** [comment]
To be fixed in #1942
